src/main/scala/mesosphere/marathon/core/group/GroupManager.scala

```diff
@@ -124,12 +125,22 @@ trait GroupManager {
     *              one can control, to stop a current deployment and start a new one.
     * @return the deployment plan which will be executed.
     */
-  def updateRoot(
+  final def updateRoot(
     id: PathId,
     fn: RootGroup => RootGroup,
     version: Timestamp = Group.defaultVersion,
     force: Boolean = false,
-    toKill: Map[PathId, Seq[Instance]] = Map.empty): Future[DeploymentPlan]
+    toKill: Map[PathId, Seq[Instance]] = Map.empty): Future[DeploymentPlan] = {
+    updateRootEither[Nothing](id, fn.andThen(Right(_)), version, force, toKill)
+      .map(_.right.getOrElse(throw new RuntimeException("Either must be Right here")))(ExecutionContexts.callerThread)
+  }
+
+  def updateRootEither[T](
+    id: PathId,
+    fn: RootGroup => Either[T, RootGroup],
```

Comment:
What's the benefit over exceptions? I seems we inject `Rejections` into the group manager with this.